pkg/proxy/accesslog/log.go

```diff
@@ -78,15 +78,20 @@ type LogRecordNotifier interface {
 }
 
 // OpenLogfile opens a file for logging
-func OpenLogfile(lf string, n LogRecordNotifier) error {
+func OpenLogfile(lf string) error {
 	logMutex.Lock()
 	defer logMutex.Unlock()
 
-	notifier = n
```

Comment:
What was the motivation to make this a package global? It makes the relationship between OpenLogFile and the notifier more hidden and indirect. Perhaps there is another PR where we'll want to change the global notifier?
If you really want to keep it this way could you also make the comment on OpenLogfile explain what it does?

Reply:
Happy about such a PR, this PR just splits the init code.